Lib9c/Action/HackAndSlashRandomBuff.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bencodex.Types;
+using Libplanet;
+using Libplanet.Action;
+using Libplanet.Assets;
+using Nekoyume.Battle;
+using Nekoyume.Helper;
+using Nekoyume.Model.State;
+using Nekoyume.TableData.Crystal;
+
+namespace Nekoyume.Action
+{
+    /// <summary>
+    /// Created at https://github.com/planetarium/lib9c/pull/1031
+    /// </summary>
+    [Serializable]
+    [ActionType("hack_and_slash_random_buff")]
+    public class HackAndSlashRandomBuff : GameAction
+    {
+        public enum BuffRank
+        {
+            SS = 1,
+            S = 2,
+            A = 3,
+            B = 4,
+        }
+
+        public const int MinimumGachaCount = 5;
+        public const int MaximumGachaCount = 10;
+
+        public Address AvatarAddress;
+        public int GachaCount;
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal
+            => new Dictionary<string, IValue>
+            {
+                ["a"] = AvatarAddress.Serialize(),
+                ["c"] = GachaCount.Serialize(),
+            }.ToImmutableDictionary();
+        protected override void LoadPlainValueInternal(IImmutableDictionary<string, IValue> plainValue)
+        {
+            AvatarAddress = plainValue["a"].ToAddress();
+            GachaCount = plainValue["c"].ToInteger();
+        }
+
+        public override IAccountStateDelta Execute(IActionContext context)
+        {
+            var states = context.PreviousStates;
+            var gachaStateAddress = AvatarAddress.Derive(Lib9c.SerializeKeys.HackAndSlashBuffStateKey);
+
+            // Invalid Avatar address, or does not have GachaState.
+            if (!states.TryGetState(gachaStateAddress, out List rawGachaState))
+            {
+                throw new FailedLoadStateException(
+                    $"Can't find {nameof(HackAndSlashBuffState)}. Gacha state address:{gachaStateAddress}");
+            }
+
+            var gachaState = new HackAndSlashBuffState(gachaStateAddress, rawGachaState);
+            var stageBuffSheet = states.GetSheet<CrystalStageBuffGachaSheet>();
+
+            // Insufficient gathered star.
+            if (gachaState.StarCount < stageBuffSheet[gachaState.StageId].MaxStar)
+            {
+                throw new NotEnoughGatheredStarException(
+                    $"Not enough gathered stars. Need : {stageBuffSheet[gachaState.StageId].MaxStar}, own : {gachaState.StarCount}");
+            }
+
+            // Invalid Gacha count.
+            if (GachaCount != MinimumGachaCount && GachaCount != MaximumGachaCount)
```

Comment:
```suggestion
            if (GachaCount != MinimumGachaCount || GachaCount != MaximumGachaCount)
```

Reply:
The sugesstion does not look good.
If change `and` operator to `or`, this if statement will be always excute.